### ISSUE TITLE
[0.10] Beta Server crashed 

### ISSUE THREAD

**Issue by SkyRocknRoll** [influxdata/influxdb#5433]
Panic occurred . 
I have migrated 0.9.x to 0.10x 
Please find the details below
### Os details 
`
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.2 LTS
Release:	14.04
Codename:	trusty
`

` influxdb                                        0.10.0-0.beta1               amd64
`
Full log Dump  -- 2.3M 
http://paste.ubuntu.com/14628206/

**Comment by toddboom** [comment]
@SkyRocknRoll thanks for the report! i just wanted to confirm a few things:

* it looks like you're running a single node, with telegraf as the only data writer. is that correct?
* what version of 0.9.x were you running prior to the upgrade?
* can you also try this with the `0.10.0-beta2` release that just came out yesterday?

**Comment by SkyRocknRoll** [comment]
Hi @toddboom , 
Thanks for looking at the issue . 
Yes telegraf is the only writer 
I am actually not sure of which minor version 0.9 i was using. But it was the latest
I will try beta2 today and let you know the status tomorrow

**Comment by SkyRocknRoll** [comment]
I have installed 0.10-beta2 and receiving the following error 
```
[cluster] 2016/01/24 15:14:59 accept remote connection from 127.0.0.1:46411
[http] 2016/01/24 15:14:59 10.0.12.11 - - [24/Jan/2016:15:14:59 +0000] POST /write?consistency=&db=telegraf&precision=s&rp= HTTP/1.1 204 0 - InfluxDBClient 3c0272e6-c2ad-11e5-8180-000000000000 142.588257ms
[http] 2016/01/24 15:14:59 10.0.0.203 - - [24/Jan/2016:15:14:59 +0000] POST /write?consistency=&db=telegraf&precision=s&rp= HTTP/1.1 204 0 - InfluxDBClient 3c037d63-c2ad-11e5-8187-000000000000 135.924764ms
[cluster] 2016/01/24 15:14:59 accept remote connection from 127.0.0.1:46410
[http] 2016/01/24 15:14:59 10.0.4.95 - - [24/Jan/2016:15:14:59 +0000] POST /write?consistency=&db=telegraf&precision=s&rp= HTTP/1.1 204 0 - InfluxDBClient 3c0387fc-c2ad-11e5-8189-000000000000 140.726169ms
[cluster] 2016/01/24 15:14:59 accept remote connection from 127.0.0.1:46407
[cluster] 2016/01/24 15:14:59 accept remote connection from 127.0.0.1:46412
[cluster] 2016/01/24 15:14:59 accept remote connection from 127.0.0.1:46413
[http] 2016/01/24 15:14:59 10.0.11.34 - - [24/Jan/2016:15:14:59 +0000] POST /write?consistency=&db=telegraf&precision=s&rp= HTTP/1.1 204 0 - InfluxDBClient 3c0372dc-c2ad-11e5-8185-000000000000 147.898358ms
[cluster] 2016/01/24 15:14:59 close remote connection from 127.0.0.1:46403
[cluster] 2016/01/24 15:14:59 close remote connection from 127.0.0.1:46404
[cluster] 2016/01/24 15:14:59 close remote connection from 127.0.0.1:46378
```

**Comment by SkyRocknRoll** [comment]
Some of the data is not appearing in the grafana graphs.

**Comment by rossmcdonald** [comment]
@SkyRocknRoll Can you show us the output to a `SHOW SERVERS` command?

**Comment by rossmcdonald** [comment]
@SkyRocknRoll Also, which storage engine were you using prior to the upgrade? It looks like this may be tsm related:

```
panic: close of closed channel

goroutine 157 [running]:
github.com/influxdb/influxdb/tsdb/engine/tsm1.(*Engine).Close(0xc208130780, 0x0, 0x0)
	/tmp/tmp.TcaAgGugKM/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/engine.go:147 +0x3e
github.com/influxdb/influxdb/tsdb.(*Shard).close(0xc2080c98c0, 0x0, 0x0)
	/tmp/tmp.TcaAgGugKM/src/github.com/influxdb/influxdb/tsdb/shard.go:150 +0x54
github.com/influxdb/influxdb/tsdb.(*Shard).Close(0xc2080c98c0, 0x0, 0x0)
	/tmp/tmp.TcaAgGugKM/src/github.com/influxdb/influxdb/tsdb/shard.go:145 +0x89
github.com/influxdb/influxdb/tsdb.(*Store).DeleteShard(0xc208098140, 0x2a, 0x0, 0x0)
	/tmp/tmp.TcaAgGugKM/src/github.com/influxdb/influxdb/tsdb/store.go:134 +0xe4
github.com/influxdb/influxdb/services/retention.(*Service).deleteShards(0xc20805ea80)
	/tmp/tmp.TcaAgGugKM/src/github.com/influxdb/influxdb/services/retention/service.go:124 +0x46b
created by github.com/influxdb/influxdb/services/retention.(*Service).Open
	/tmp/tmp.TcaAgGugKM/src/github.com/influxdb/influxdb/services/retention/service.go:46 +0x1b8
```

**Comment by SkyRocknRoll** [comment]
i have used default one which comes 0.9.x.

**Comment by SkyRocknRoll** [comment]
I have wiped the whole data and started fresh server. Now influxdb is working fine for last 1 hour. Will keep update the progress

**Comment by dswarbrick** [comment]
I noticed missing data in graphs also, when upgrading from an existing nightly (0.10.0.n1453215519) to the latests. It seemed to only affect derivative queries, and after nuking the DB and waiting for some new data to accumulate, the derivative queries worked fine again.

**Comment by dswarbrick** [comment]
I also just noticed that 0.10.0-beta2 appears to hang indefinitely when performing a level 3 compaction, which for me is about 30 minutes after starting influxd. I am downgrading to 0.10.0-beta1 again.

**Comment by rossmcdonald** [comment]
@dswarbrick Thanks for letting us know. Was there any extra information (log messages, etc) provided prior to or during the hang? And, to clarify, the hangs occurred with a clean database?

If you see it happen again, it would be great if you could create a new issue with the output of a trace generated by a SIGQUIT (`kill -SIGQUIT $(pgrep influxd)`) while it's hung.

**Comment by dswarbrick** [comment]
@rossmcdonald These were the last log messages before I shut down influxd:
```
[tsm1] 2016/01/24 19:07:44 beginning level 1 compaction of group 0, 2 TSM files
[tsm1] 2016/01/24 19:07:44 compacting level 1 group (0) /srv/influxdb/data/perfdb/default/2/000000015-000000001.tsm (#0)
[tsm1] 2016/01/24 19:07:44 compacting level 1 group (0) /srv/influxdb/data/perfdb/default/2/000000016-000000001.tsm (#1)
[query] 2016/01/24 19:07:53 SELECT non_negative_derivative(mean(pointReq), 1s) FROM _internal.monitor."write" WHERE time > now() - 30m GROUP BY time(1m)
[tsm1] 2016/01/24 19:07:58 compacted level 1 group (0) into /srv/influxdb/data/perfdb/default/2/000000016-000000002.tsm.tmp (#0)
[tsm1] 2016/01/24 19:07:58 compacted level 1 group 0 of 2 files into 1 files in 13.631551827s
[tsm1] 2016/01/24 19:07:59 beginning level 2 compaction of group 0, 2 TSM files
[tsm1] 2016/01/24 19:07:59 compacting level 2 group (0) /srv/influxdb/data/perfdb/default/2/000000014-000000002.tsm (#0)
[tsm1] 2016/01/24 19:07:59 compacting level 2 group (0) /srv/influxdb/data/perfdb/default/2/000000016-000000002.tsm (#1)
[tsm1] 2016/01/24 19:08:17 compacted level 2 group (0) into /srv/influxdb/data/perfdb/default/2/000000016-000000003.tsm.tmp (#0)
[tsm1] 2016/01/24 19:08:17 compacted level 2 group 0 of 2 files into 1 files in 18.35735472s
[tsm1] 2016/01/24 19:08:17 beginning level 3 compaction of group 0, 4 TSM files
[tsm1] 2016/01/24 19:08:17 compacting level 3 group (0) /srv/influxdb/data/perfdb/default/2/000000004-000000003.tsm (#0)
[tsm1] 2016/01/24 19:08:17 compacting level 3 group (0) /srv/influxdb/data/perfdb/default/2/000000008-000000003.tsm (#1)
[tsm1] 2016/01/24 19:08:17 compacting level 3 group (0) /srv/influxdb/data/perfdb/default/2/000000012-000000003.tsm (#2)
[tsm1] 2016/01/24 19:08:17 compacting level 3 group (0) /srv/influxdb/data/perfdb/default/2/000000016-000000003.tsm (#3)
[query] 2016/01/24 19:08:53 SELECT non_negative_derivative(mean(pointReq), 1s) FROM _internal.monitor."write" WHERE time > now() - 30m GROUP BY time(1m)
[tsm1] 2016/01/24 19:10:01 compacted level 3 group (0) into /srv/influxdb/data/perfdb/default/2/000000016-000000004.tsm.tmp (#0)
[tsm1] 2016/01/24 19:10:01 compacted level 3 group 0 of 4 files into 1 files in 1m43.788494609s
[run] 2016/01/24 19:15:40 Signal received, initializing clean shutdown...
[run] 2016/01/24 19:15:40 Waiting for clean shutdown...
[copier] 2016/01/24 19:15:40 copier listener closed
[snapshot] 2016/01/24 19:15:40 snapshot listener closed
[cluster] 2016/01/24 19:15:40 cluster service accept error: network connection closed
[run] 2016/01/24 19:15:40 second signal received, initializing hard shutdown
[shard-precreation] 2016/01/24 19:15:40 Precreation service terminating
```
After 19:10, I noticed that Grafana graphs were not updating (perpetual "loading" spinner), and I was not able to connect using the influx CLI. Netstat showed dozens of TCP sockets' receive buffers were backing up. This was with a fresh database, created by 0.10.0-beta2.

Unfortunately I can't repeat this test on my semi-production setup, since it causes problems for the collectd daemons, which don't enforce an HTTP timeout when sending the data. I had to kill -9 several dozen collectd daemons. I have another setup, running 0.10.0-beta2, but it does not receive anywhere near the amount of data, so it may take a while before it does a level 3 compaction.

**Comment by dswarbrick** [comment]
@rossmcdonald #5437 created.

**Comment by jwilder** [comment]
The `close of closed channel` panic is a duplicate of #5244.

**Comment by rossmcdonald** [comment]
Closing in favor of #5244.
